jx_sqlite/__init__.py

```diff
@@ -211,11 +211,15 @@ def _get(row):
     return _get
 
 
-def set_column(row, col, child, value):
+def set_column(row, col, name, child, value, header):
     if child == ".":
         row[col] = value
     else:
         column = row[col]
+
+        if column is None and child in header[col]:
+            column = {}
```

Comment:
the child should never end up in the header.  The child is meant to describe the structure inside a single column, the header gives names to the columns. 

Traditionally, databases demanded that all columns be primitive values, this is not the case whe JSON query expressions:  A column can be JSON document, and that JSON document is described by the `c.push_child` from all the columns.